### ISSUE TITLE
pn.state.schedule_task repeats many times for one call?

### ISSUE THREAD

**Issue by tomascsantos** [holoviz/panel#3794]
#### ALL software version info
(this library, plus any other relevant software, e.g. bokeh, python, notebook, OS, browser, etc)
https://gist.github.com/tomascsantos/40d5575d1faab94960c2be673337bcca

#### Description of expected behavior and the observed behavior
When I pass a simple script to `--setup warm_cache.py` I expect the invoked function to be called ONCE immediately, and then once each 45 minutes after this, instead it seems to be called SEVEN times immediately?

#### Complete, minimal, self-contained example code that reproduces the issue

```python
#warm_cache.py
print('warm cache started')
import panel as pn

def test():
    print('we do test')

# Update cache each day at 6am PST
print('we schedule warm cache task')
pn.state.schedule_task(
    name='warm_cache',
    callback=test,
    cron='45 * * * *',
)
```

#### Screenshots of the bug in action
![image](https://user-images.githubusercontent.com/7690514/187253501-9c4da29d-f49d-46de-a11e-095671c8d8e2.png)

**Comment by z--m-n** [comment]
Experienced something similar, but could  get the scheduling in`--setup app_setup.py` working using the `at=` and `period=` option to `pn.state.schedule_task(...)` , instead of `cron=`. 

Similar to cron, the `at=` time can be set as the next start of a period, counting from epoch. Cron is more flexible, so this is more a lukewarm cache scheduling, but it works.

**Comment by ledwithQ** [comment]
@tomascsantos I came across this issue today. I would imagine we feel the issue more directly coming from North American time zones and the issue seems to be only on startup.  I believe there is a bug in when panel seeds the croniter expression.

https://github.com/holoviz/panel/blob/main/panel/io/state.py#L856
I believe the below needs to be written should  replace `dt.datetime.now()` with `dt.datetime.utcnow()` 

```from croniter import croniter
 base = dt.datetime.now() if at is None else at
 diter = croniter(cron, base)
```

**As a workaround:**  We are getting the desired startup sequence when we include the `at` argument with the cron expression as `at=dt.datetime.utcnow()`

**Comment by philippjfr** [comment]
@ledwithQ Looks like you have a decent handle on what is wrong, any chance you could contribute a fix?

**Comment by Hoxbro** [comment]
Note that Python 3.12 has deprecated `dt.datetime.utcnow()`.

**Comment by Hoxbro** [comment]
I agree with you and would use `dt.datetime.now(tz=dt.timezone.utc)`.

**Comment by philippjfr** [comment]
Also agree, let's work with UTC whenever possible.
